packages/react-dom/src/test-utils/ReactTestUtils.js

```diff
@@ -389,58 +387,68 @@ const ReactTestUtils = {
 
   Simulate: null,
   SimulateNative: {},
-
-  act(callback: () => void): Thenable {
+  act(callback: () => void | Promise<void>) {
     if (actContainerElement === null) {
-      // warn if we can't actually create the stub element
       if (__DEV__) {
-        warningWithoutStack(
-          typeof document !== 'undefined' &&
-            document !== null &&
-            typeof document.createElement === 'function',
-          'It looks like you called TestUtils.act(...) in a non-browser environment. ' +
-            "If you're using TestRenderer for your tests, you should call " +
-            'TestRenderer.act(...) instead of TestUtils.act(...).',
-        );
+        // warn if we're trying to use this in something like node (without jsdom)
+        if (didWarnAboutActInNodejs === false) {
+          didWarnAboutActInNodejs = true;
+          warningWithoutStack(
+            typeof document !== 'undefined' && document !== null,
+            'It looks like you called ReactTestUtils.act(...) in a non-browser environment. ' +
+              "If you're using TestRenderer for your tests, you should call " +
+              'ReactTestRenderer.act(...) instead of ReactTestUtils.act(...).',
+          );
+        }
       }
-      // then make it
+      // now make the stub element
       actContainerElement = document.createElement('div');
     }
-
-    const result = ReactDOM.unstable_batchedUpdates(callback);
     // note: keep these warning messages in sync with
     // createReactNoop.js and ReactTestRenderer.js
-    if (__DEV__) {
-      if (result !== undefined) {
-        let addendum;
-        if (result !== null && typeof result.then === 'function') {
-          addendum =
-            '\n\nIt looks like you wrote ReactTestUtils.act(async () => ...), ' +
-            'or returned a Promise from the callback passed to it. ' +
-            'Putting asynchronous logic inside ReactTestUtils.act(...) is not supported.\n';
-        } else {
-          addendum = ' You returned: ' + result;
-        }
-        warningWithoutStack(
-          false,
-          'The callback passed to ReactTestUtils.act(...) function must not return anything.%s',
-          addendum,
-        );
+    const result = actedUpdates(callback);
+    if (
+      result !== null &&
+      typeof result === 'object' &&
+      typeof result.then === 'function'
+    ) {
+      let called = false;
+      if (__DEV__) {
+        // eslint-disable-next-line no-undef
+        Promise.resolve()
```

Comment:
Technically we don't require promise support. Since this is just a warning, wrap it in an existence check.

Reply:
Getting a particularly odd error when I put in the existence check (see the CI error.) posted about it internally, let's see.

Reply:
Which CI error? And I can't find the internal post. Send to me via Work Chat? :D

Reply:
alright, the current version seems to appease both lint and flow. marking this resolved.